eclipse-theia/theia
# Support `DocumentDropEditProvider` (#12025)

#### What it does
resolves #12025
this implements the vscode extension api for 
- `DocumentDropEdit `
- `DocumentDropEditProvider` 
- as well as the corresponding registration method in the `language` namespace

also implments DTOs including converters and ability to resolve the file Data for those for #11776.

#### How to test
here is a small test extension: 
[documentdropedit-test-0.0.1.zip](https://github.com/eclipse-theia/theia/files/10536306/documentdropedit-test-0.0.1.zip)

or you can build it here:
https://github.com/jonah-iden/documentdropedit-test

1. install extension
2. execute "Register dropEdit provider"
3. drop one or multiple fils into editor
4. should see messages with name of each file and content of one file should have been added to current cursor position in the editor

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
